File: Programacion multimedia y dispositivos moviles2T/GestorDeTareas/GestorDeTareas/app/src/main/java/com/example/gestordetareas/vista/RegistroActivity.java
package com.example.gestordetareas.vista;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.gestordetareas.R;
import com.example.gestordetareas.dao.UsuarioDAO;
import com.example.gestordetareas.utils.EncriptacionUtils;

public class RegistroActivity extends AppCompatActivity {

    private EditText etNombre, etCorreo, etPassword, etTelefono, etCiudad;
    private UsuarioDAO usuarioDAO;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registro);

        // Inicializar vistas
        etNombre = findViewById(R.id.et_nombre);
        etCorreo = findViewById(R.id.et_correo);
        etPassword = findViewById(R.id.et_password);
        etTelefono = findViewById(R.id.et_telefono);
        etCiudad = findViewById(R.id.et_ciudad);
        Button btnRegistrar = findViewById(R.id.btn_registrar);
        Button btnVolver = findViewById(R.id.btn_volver);

        // Inicializar DAO
        usuarioDAO = new UsuarioDAO(this);

        // Botón Registrar
        btnRegistrar.setOnClickListener(v -> registrarUsuario());

        // Botón Volver
        btnVolver.setOnClickListener(v -> finish());
    }

    private void registrarUsuario() {
        String nombre = etNombre.getText().toString().trim();
        String correo = etCorreo.getText().toString().trim();
        String password = etPassword.getText().toString().trim();
        String telefono = etTelefono.getText().toString().trim();
        String ciudad = etCiudad.getText().toString().trim();

        // Validación de campos vacíos
        if (nombre.isEmpty() || correo.isEmpty() || password.isEmpty() || telefono.isEmpty() || ciudad.isEmpty()) {
            Toast.makeText(this, "⚠️ Todos los campos son obligatorios", Toast.LENGTH_SHORT).show();
            return;
        }

        // Verificar si el usuario ya existe
        if (usuarioDAO.verificarLogin(correo, "")) {  // Solo verificamos si existe el usuario
            Toast.makeText(this, "⚠️ El correo ya está registrado", Toast.LENGTH_SHORT).show();
            return;
        }


        // Encriptar contraseña
        String passwordEncriptada = EncriptacionUtils.encriptar(password);

        // Registrar usuario
        boolean registroExitoso = usuarioDAO.registrarUsuario(nombre, correo, passwordEncriptada, telefono, ciudad);

        if (registroExitoso) {
            Toast.makeText(this, "✅ Registro exitoso. Inicia sesión.", Toast.LENGTH_LONG).show();
            startActivity(new Intent(RegistroActivity.this, LoginActivity.class));
            finish();
        } else {
            Toast.makeText(this, "❌ Error al registrar usuario", Toast.LENGTH_SHORT).show();
        }
    }
}

File: Programacion multimedia y dispositivos moviles2T/GestionDeStock/GestionDeStock/app/src/main/java/com/example/gestiondestock/activities/RegistroActivity.kt
package com.example.gestiondestock.activities

import android.content.ContentValues
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.gestiondestock.R
import com.example.gestiondestock.database.BaseDatosHelper

class RegistroActivity : AppCompatActivity() {

    private lateinit var baseDatosHelper: BaseDatosHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registro)

        baseDatosHelper = BaseDatosHelper(this)

        val campoUsuario = findViewById<EditText>(R.id.editTextUsuarioRegistro)
        val campoContrasena = findViewById<EditText>(R.id.editTextContrasenaRegistro)
        val botonRegistrar = findViewById<Button>(R.id.btnRegistrarUsuario)

        botonRegistrar.setOnClickListener {
            val usuario = campoUsuario.text.toString()
            val contrasena = campoContrasena.text.toString()

            if (usuario.isNotBlank() && contrasena.isNotBlank()) {
                val db = baseDatosHelper.writableDatabase
                val cursor = db.rawQuery(
                    "SELECT * FROM usuarios WHERE nombre = ?",
                    arrayOf(usuario)
                )
                if (cursor.count > 0) {
                    Toast.makeText(this, "El usuario ya existe", Toast.LENGTH_SHORT).show()
                    cursor.close()
                } else {
                    cursor.close()
                    val valores = ContentValues().apply {
                        put("nombre", usuario)
                        put("contrasena", contrasena)
                    }
                    db.insert("usuarios", null, valores)
                    Toast.makeText(this, "Usuario registrado correctamente", Toast.LENGTH_SHORT).show()
                    finish()
                }
            } else {
                Toast.makeText(this, "Por favor, completa todos los campos", Toast.LENGTH_SHORT).show()
            }
        }
    }
}
